File: src/test/java/io/atesfactory/evrl/ResourceFactoryTests.java
package io.atesfactory.evrl;

import io.atesfactory.evrl.configuration.EvrlSpringContext;
import io.atesfactory.evrl.resourcefactory.ResourceFactoryContext;
import io.atesfactory.evrl.resourcefactory.ResourceFactoryRegistry;
import java.io.File;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.mockito.Mockito;
import org.springframework.context.ApplicationContext;
import org.springframework.core.env.Environment;
import org.springframework.core.io.FileSystemResource;
import org.springframework.core.io.InputStreamResource;

class ResourceFactoryTests {

    private final EvrlSpringContext evrlSpringContext = new EvrlSpringContext(
            Mockito.mock(ApplicationContext.class),
            Mockito.mock(Environment.class)
    );

    @Test
    void shouldCreateInputStreamResourceFromByteArray() {
        byte[] content = "Hello".getBytes();

        Object resource = ResourceFactoryRegistry.get(content.getClass()).create(
                new ResourceFactoryContext(evrlSpringContext), content);

        Assertions.assertEquals(InputStreamResource.class, resource.getClass());
    }

    @Test
    void shouldCreateFileSystemResourceFromFile() {
        File content = new File("myFile");

        Object resource = ResourceFactoryRegistry.get(content.getClass()).create(
                new ResourceFactoryContext(evrlSpringContext), content);

        Assertions.assertEquals(FileSystemResource.class, resource.getClass());
    }
}

File: src/test/kotlin/io/atesfactory/evrl/ResourceFactoryTests.kt
package io.atesfactory.evrl

import io.atesfactory.evrl.configuration.EvrlSpringContext
import io.atesfactory.evrl.resourcefactory.ResourceFactoryContext
import io.atesfactory.evrl.resourcefactory.ResourceFactoryRegistry
import java.io.File
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.mockito.Mockito
import org.springframework.context.ApplicationContext
import org.springframework.core.env.Environment
import org.springframework.core.io.FileSystemResource
import org.springframework.core.io.InputStreamResource

class ResourceFactoryTests {

    private var evrlSpringContext = EvrlSpringContext(
        Mockito.mock(ApplicationContext::class.java),
        Mockito.mock(Environment::class.java)
    )

    @Test
    fun `should create InputStreamResource from ByteArray`(){
        val content = "Hello".toByteArray()

        val resource = ResourceFactoryRegistry.get(content::class.java).create(
            ResourceFactoryContext(evrlSpringContext), content)

        assertEquals(InputStreamResource::class.java, resource::class.java)
    }

    @Test
    fun `should create FileSystemResource from File`() {
        val content = File("myFile")

        val resource = ResourceFactoryRegistry.get(content::class.java).create(
            ResourceFactoryContext(evrlSpringContext), content)

        assertEquals(FileSystemResource::class.java, resource::class.java)
    }

}
